api/src/main/java/ai/djl/ndarray/index/NDIndex.java

```diff
@@ -373,7 +373,7 @@ private int addIndexItem(String indexItem, int argIndex, Object[] args) {
                     if (array.getDataType() == DataType.BOOLEAN) {
                         indices.add(new NDIndexBooleans(array));
                         return argIndex + 1;
-                    } else if (array.getDataType().isInteger()) {
```

Comment:
Why our integration test didn't catch this issue?
Can we add more test to ensure this not happen again?

Reply:
+1 we can add the case from the bug that reproduces this as part of our tests

Reply:
It was ot caught since previously the tests only tested integer index arrays. Now I have added float index array as well.